test/test_components.py

```diff
@@ -0,0 +1,211 @@
+import numpy as np
+import pytest
+
+import pypsa
+
+
+@pytest.fixture
+def network():
+    return pypsa.examples.ac_dc_meshed(from_master=True)
+
+
+@pytest.fixture
+def empty_network_5_buses():
+    # Set up empty network with 5 buses.
+    network = pypsa.Network()
+    n_buses = 5
+    for i in range(n_buses):
+        network.add(
+            'Bus',
+            f'bus_{i}'
+        )
+    return network
+
+
+def test_mremove(network):
+    """
+    GIVEN   the AC DC exemplary pypsa network
+
+    WHEN    two components of Generator are removed with mremove
+
+    THEN    the generator dataframe and the time-dependent generator dataframe
+                should not contain the removed elements.
+    """
+    generators = ['Manchester Wind', 'Frankfurt Wind']
+
+    network.mremove('Generator', generators)
+
+    assert generators not in network.generators.index.tolist()
+    assert generators not in network.generators_t.p_max_pu.columns.tolist()
```

Comment:
```suggestion
    generators = {'Manchester Wind', 'Frankfurt Wind'}

    network.mremove('Generator', generators)

    assert not generators.issubset(network.generators.index)
    assert not generators.issubset(network.generators_t.p_max_pu.columns)
```
Hey @rockstaedt, just thought a bit about the structure, I'd prefer to work with sets here. Could you double-check my suggestion and if it works fine apply the pattern to the other functions?